components/slack/sources/common/base.mjs

```diff
@@ -0,0 +1,84 @@
+import slack from "../../slack.app.mjs";
+
+export default {
+  props: {
+    slack,
+    nameCache: "$.service.db",
+  },
+  methods: {
+    ...slack.methods,
+    async maybeCached(key, refreshVal, timeoutMs = 3600000) {
+      let record = this.nameCache.get(key);
+      const time = Date.now();
+      if (!record || time - record.ts > timeoutMs) {
+        record = {
+          ts: time,
+          val: await refreshVal(),
+        };
+        this.nameCache.set(key, record);
+      }
+      return record.val;
+    },
+    async getUserName(id) {
+      return this.maybeCached(`users:${id}`, async () => {
+        const info = await this.slack.sdk().users.info({
+          user: id,
+        });
+        if (!info.ok) throw new Error(info.error);
+        return info.user.name;
+      });
+    },
+    async getBotName(id) {
+      return this.maybeCached(`bots:${id}`, async () => {
+        const info = await this.slack.sdk().bots.info({
+          bot: id,
+        });
+        if (!info.ok) throw new Error(info.error);
+        return info.bot.name;
+      });
+    },
+    async getConversationName(id) {
+      return this.maybeCached(`conversations:${id}`, async () => {
+        const info = await this.slack.sdk().conversations.info({
+          channel: id,
+        });
+        if (!info.ok) throw new Error(info.error);
+        if (info.channel.is_im) {
+          return `DM with ${await this.getUserName(info.channel.user)}`;
+        } else {
+          return info.channel.name;
+        }
```

Comment:
This can be simplified:
```suggestion
        if (info.channel.is_im) {
          return `DM with ${await this.getUserName(info.channel.user)}`;
        }
        return info.channel.name;
```